src/BizHawk.Client.Common/Api/Interfaces/IMemoryApi.cs

```diff
@@ -17,6 +17,15 @@ public interface IMemoryApi : IExternalApi
 
 		uint ReadByte(long addr, string domain = null);
 		IReadOnlyList<byte> ReadByteRange(long addr, int length, string domain = null);
+
+		/// <summary>copies a region of memory into <paramref name="dstBuffer"/>, starting at <paramref name="srcStartOffset"/> and taking as many bytes as will fit in the buffer</summary>
+		/// <returns>
+		/// <see langword="false"/> iff the internal API call threw an exception (for example, if you request bytes beyond the end of the domain).<br/>
+		/// If this happens, the state of <paramref name="dstBuffer"/> is undefined. Anywhere from <c>0</c> bytes to all of them could have been copied.
+		/// The exception is when <paramref name="srcStartOffset"/> falls outside the domain, as that is checked first and results in a no-op.
+		/// </returns>
+		bool ReadByteRange(ulong srcStartOffset, Span<byte> dstBuffer, string domain = null);
```

Comment:
The doc comment here is wrong. It returns false if the internal API call threw an exception, but not iff: the method itself contains a check for when the start falls outside the domain.
This is also different behavior from the other overload, which will still return some bytes just starting at the beginning of the memory domain. I think they should have the same behavior. (which again IMO should be throwing)